cognite/neat/graph/extractors/_iodd.py

```diff
@@ -0,0 +1,141 @@
+import xml.etree.ElementTree as ET
+from pathlib import Path
+from xml.etree.ElementTree import Element
+
+from rdflib import RDF, Literal, Namespace, URIRef
+
+from cognite.neat.constants import DEFAULT_NAMESPACE
+from cognite.neat.graph.extractors._base import BaseExtractor
+from cognite.neat.graph.models import Triple
+from cognite.neat.utils.rdf_ import as_neat_compliant_uri
+from cognite.neat.utils.text import to_camel
+from cognite.neat.utils.xml_ import iterate_tree, remove_element_tag_namespace
+
+IODD = Namespace("http://www.io-link.com/IODD/2010/10/")
+XSI = Namespace("http://www.w3.org/2001/XMLSchema-instance/")
+
+
+class IODDExtractor(BaseExtractor):
+    """
+    IODD-XML extractor of RDF triples
+
+    Each IODD sheet describes an IODD device. We want to extract triples that describes the device, and the
+    sensors connected to the device.
+    This data is described under the elements "DeviceIdentity" and "ProcessDataCollection".
+    We can also extend the IODDExtractor to extract triples from other elements related to the device such as
+    "EventCollection".
+
+    Args:
+        root: XML root element of IODD XML file.
+        namespace: Optional custom namespace to use for extracted triples that define data
+                    model instances. Defaults to DEFAULT_NAMESPACE.
+    """
+
+    def __init__(self, root: Element, namespace: Namespace | None = None):
+        self.root = root
+        self.namespace = namespace or DEFAULT_NAMESPACE
+
+    @classmethod
+    def from_file(cls, filepath: str | Path, namespace: Namespace | None = None):
+        return cls(ET.parse(filepath).getroot(), namespace)
+
+    @classmethod
+    def _element2triples(cls, root: Element) -> list[Triple]:
+        """Converts an element to triples."""
+        triples: list[Triple] = []
+
+        # Extract DeviceIdentity triples
+        if di_root := root.find(".//{*}DeviceIdentity"):
```

Comment:
Since you have now improved version of `get_children` method remove need to do `.find` in iodd extractor and leverage on the method you created instead